src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/PersonalizeRankingResponseProcessor.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.search.relevance.transformer.personalizeintelligentranking;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.SearchResponseSections;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.search.SearchHits;
+import org.opensearch.search.aggregations.InternalAggregations;
+import org.opensearch.search.internal.InternalSearchResponse;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+import org.opensearch.search.profile.SearchProfileShardResults;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.client.PersonalizeClient;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.client.PersonalizeClientSettings;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.client.PersonalizeCredentialsProviderFactory;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.configuration.PersonalizeIntelligentRankerConfiguration;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.requestparameter.PersonalizeRequestParameterUtil;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.requestparameter.PersonalizeRequestParameters;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.reranker.PersonalizedRanker;
+import org.opensearch.search.relevance.transformer.personalizeintelligentranking.reranker.PersonalizedRankerFactory;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+
+import static org.opensearch.search.relevance.transformer.personalizeintelligentranking.configuration.Constants.CAMPAIGN_ARN_CONFIG_NAME;
```

Comment:
it seems that the class is importing individual configuration constants from the configuration package. While this approach can work, I believe it might be more maintainable and flexible to incorporate the configuration as an attribute of the plugins and access it using `this.personalizeClientsetting`. This way, the class can receive the configuration as a parameter, allowing for greater modularity and easier testing.

Check out this PR, which use `this.kendraClientsetting`
https://github.com/opensearch-project/search-processor/blob/01ca2045eca5bc1c5ebba7a552e362738c1868be/src/main/java/org/opensearch/search/relevance/SearchRelevancePlugin.java#L95

https://github.com/opensearch-project/search-processor/blob/01ca2045eca5bc1c5ebba7a552e362738c1868be/src/main/java/org/opensearch/search/relevance/SearchRelevancePlugin.java#L119

Reply:
Hi mingshl, 
Thank you for your review and comment.

I kept code for getting ``PersonalizeClientSettings`` instance to limited to the ``PersonalizeRankingResponseProcessor.Factory`` instance creation so that ``PersonalizeClientSettings`` are not created when they are not required. For e.g. Creating that instance outside factory creation may create  ``PersonalizeClientSettings`` instance when using plugin for Kendra only use case.

I did thought about creating one common class for Kendra and Personalize related settings. But Kendra needs more configurations than just credentials and hence opted to keep separate class for Personalize.

Let me know your thoughts regarding above.

Reply:
Makes sense. Updated code to move search pipeline related configuration constants to Factory class instead of using earlier approach of global constants.